File: version-app/src/main/java/io/ehdev/version/service/version/VersionCreateModel.java
package io.ehdev.version.service.version;

import java.util.List;

public class VersionCreateModel extends VersionSearchModel {
    String message;

    public VersionCreateModel(List<String> commits, String message) {
        super(commits);
        this.message = message;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}

File: api/rest-model/src/main/kotlin/io/ehdev/conrad/model/version/VersionCreateModel.kt
package io.ehdev.conrad.model.version

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import javax.validation.constraints.NotNull

class VersionCreateModel public @JsonCreator constructor(
    @JsonProperty("commits") commits: List<String>,
    @NotNull @JsonProperty("message") val message: String,
    @NotNull @JsonProperty("commitId") val commitId: String
): VersionSearchModel(commits);
